src/classes/ACCT_CascadeDeleteLookups_TDTM.cls

```diff
@@ -33,51 +33,240 @@
+        public List<SObject> load(Set<Id> accountIds) {
+            List<SObject> result = new List<SObject>();
+
+            List<Contact> relatedContacts = [
+                SELECT Id
+                FROM Contact 
+                WHERE AccountId IN :accountIds
+            ];
+
+            relatedRecurringDonations = getRelatedRecurringDonations(accountIds, relatedContacts);
+            result.addAll((List<SObject>) relatedRecurringDonations);
+
+            List<npe4__Relationship__c> relatedRelationships = getRelatedRelationships(relatedContacts);
+            result.addAll((List<SObject>) relatedRelationships);
+
+            return result;
+        }
+
+        /*******************************************************************************************************
+        * @description Retrieves the next children records group to be undeleted. In this case, there is another
+        * group of children records to undelete.
+        * @return CDL_CascadeDeleteLookups.CascadeUndeleteLoader.
+        ********************************************************************************************************/
+        public CDL_CascadeDeleteLookups.CascadeUndeleteLoader next() {
+            return new SecondCascadeUndeleteLoader(relatedRecurringDonations);
+
+        }
+
+        /*******************************************************************************************************
+        * @description Retrieves the related Recurring Donations of the undeleted Accounts.
+        * @param Set<Id> The Ids of the undeleted Accounts.
+        * @param List<Contact> The Contacts of the undeleted Accounts
+        * @return List<npe03__Recurring_Donation__c> Undeleted Accounts' Recurring Donations.
+        ********************************************************************************************************/
+        private List<npe03__Recurring_Donation__c> getRelatedRecurringDonations(Set<Id> accountIds, List<Contact> relatedContacts) {
+            return [
+                SELECT Name, npe03__Organization__c, npe03__Contact__r.AccountId
+                FROM npe03__Recurring_Donation__c
+                WHERE IsDeleted = TRUE
+                AND (npe03__Organization__c IN :accountIds OR npe03__Contact__c IN :relatedContacts)
+                ALL ROWS
+            ];
+        }
+
+        /*******************************************************************************************************
+        * @description Retrieves the related Relationships of the undeleted Accounts.
+        * @param List<Contact> The Contacts of the undeleted Accounts.
+        * @return List<npe4__Relationship__c> Undeleted Accounts' Relationships.
+        ********************************************************************************************************/
+        private List<npe4__Relationship__c> getRelatedRelationships(List<Contact> relatedContacts) {
```

Comment:
this seems to be an exact duplicate of the previous getRelatedRelationships() in CascadeDeleteLoader.  why not share this method by using a private static?  I'm guessing several of your other select routines also could be shared.

Reply:
@davidhabib they are different functions. The first one returns relationships that are not deleted, and the second one return relationships that are already deleted.

Reply:
missed that.  thanks @kofijohnson